pkg/istio-agent/sds-agent.go

```diff
@@ -274,113 +342,48 @@ func (sa *Agent) newWorkloadSecretCache() (workloadSecretCache *cache.SecretCach
 	workloadSecretCache = cache.NewSecretCache(fetcher, sds.NotifyProxy, sa.secOpts)
 
 	// If proxy is using file mounted certs, we do not have to connect to CA.
+	// FILE_MOUNTED_CERTS=true
 	if sa.secOpts.FileMountedCerts {
 		log.Info("Workload is using file mounted certificates. Skipping connecting to CA")
 		return
 	}
 
 	// TODO: this should all be packaged in a plugin, possibly with optional compilation.
-	log.Infof("sa.serverOptions.CAEndpoint == %v", sa.secOpts.CAEndpoint)
+	log.Infof("sa.serverOptions.CAEndpoint == %v %s", sa.secOpts.CAEndpoint, sa.secOpts.CAProviderName)
 	if sa.secOpts.CAProviderName == "GoogleCA" || strings.Contains(sa.secOpts.CAEndpoint, "googleapis.com") {
 		// Use a plugin to an external CA - this has direct support for the K8S JWT token
 		// This is only used if the proper env variables are injected - otherwise the existing Citadel or Istiod will be
 		// used.
 		caClient, err = gca.NewGoogleCAClient(sa.secOpts.CAEndpoint, true)
 		sa.secOpts.PluginNames = []string{"GoogleTokenExchange"}
+		gcpInfo := tokenmanager.GetGCPProjectInfo()
+		if stsclient.GKEClusterURL == "" && gcpInfo.Number != "" {
+			stsclient.GKEClusterURL = gcpInfo.GKEClusterURL()
+		}
+		if stsclient.GKEClusterURL == "" {
+			log.Fatalf("Failed to start, GoogleCA requires GKE_CLUSTER_URL to be set")
+		}
+		sa.secOpts.TokenExchangers = []security.TokenExchanger{stsclient.NewPlugin()}
```

Comment:
Not quite sure how this plays into the rest of the PR?

Reply:
Removed.